src/overlays/actors/ovl_En_Mag/z_en_mag.c

```diff
@@ -27,21 +33,553 @@ const ActorInit En_Mag_InitVars = {
+            this->copyrightAlpha = 0.0f;
+            this->globalState = MAG_STATE_POST_DISPLAY;
+        }
+    }
+
+    if (this->globalState == MAG_STATE_INITIAL) {
+        if (Flags_GetEnv(globalCtx, 3)) {
+            this->effectFadeInTimer = 40;
+            this->globalState = MAG_STATE_FADE_IN;
+        }
+    } else if (this->globalState == MAG_STATE_DISPLAY) {
+        if (Flags_GetEnv(globalCtx, 4)) {
+            this->globalState = MAG_STATE_FADE_OUT;
+        }
+    }
+}
+
+void EnMag_DrawTextureI8(Gfx** gfxp, void* texture, s16 texWidth, s16 texHeight, s16 rectLeft, s16 rectTop,
+                         s16 rectWidth, s16 rectHeight, u16 dsdx, u16 dtdy) {
+    Gfx* gfx = *gfxp;
+
+    gDPLoadTextureBlock(gfx++, texture, G_IM_FMT_I, G_IM_SIZ_8b, texWidth, texHeight, 0, G_TX_NOMIRROR | G_TX_WRAP,
+                        G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+
+    gSPTextureRectangle(gfx++, rectLeft * 4, rectTop * 4, (rectLeft + rectWidth) * 4, (rectTop + rectHeight) * 4,
+                        G_TX_RENDERTILE, 0, 0, dsdx, dtdy);
+
+    *gfxp = gfx;
+}
+
+void EnMag_DrawEffectTextures(Gfx** gfxp, void* maskTex, void* effectTex, s16 maskWidth, s16 maskHeight,
+                              s16 effectWidth, s16 effectHeight, s16 rectLeft, s16 rectTop, s16 rectWidth,
+                              s16 rectHeight, u16 dsdx, u16 dtdy, u16 shifts, u16 shiftt, u16 flag, EnMag* this) {
+    Gfx* gfx = *gfxp;
+
+    gDPLoadMultiBlock_4b(gfx++, maskTex, 0x0000, 0, G_IM_FMT_I, maskWidth, maskHeight, 0, G_TX_NOMIRROR | G_TX_WRAP,
+                         G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+
+    if (1) {}
+
+    if (!flag) {
+        gDPLoadMultiBlock(gfx++, effectTex, 0x0100, 1, G_IM_FMT_I, G_IM_SIZ_8b, effectWidth, effectHeight, 0,
+                          G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, 5, shifts, shiftt);
+
+        gDPSetTileSize(gfx++, 1, 0, this->effectScroll & 0x7F, 0x7C, (this->effectScroll & 0x7F) + 0x7C);
+    }
+
+    gSPTextureRectangle(gfx++, rectLeft * 4, rectTop * 4, (rectLeft + rectWidth) * 4, (rectTop + rectHeight) * 4,
+                        G_TX_RENDERTILE, 0, 0, dsdx, dtdy);
+
+    *gfxp = gfx;
+}
+
+void EnMag_DrawImageRGBA32(Gfx** gfxp, s16 startLeft, s16 startTop, u8* source, u32 sizeX, u32 sizeY) {
+    Gfx* gfx = *gfxp;
+    u8* curTexture;
+    s32 textureCount;
+    u32 rectLeft;
+    u32 rectTop;
+    u32 height;
+    s32 remainingSize;
+    s32 textureSize;
+    s32 pad;
+    s32 i;
+
+    func_80094D28(&gfx);
+
+    curTexture = source;
+    rectLeft = startLeft - (sizeX >> 1);
+    rectTop = startTop - (sizeY >> 1);
+    height = 4096 / (sizeX << 2);
+    remainingSize = (sizeX * sizeY) << 2;
+    textureSize = (sizeX * height) << 2;
+    textureCount = remainingSize / textureSize;
+    if ((remainingSize % textureSize) != 0) {
+        textureCount++;
+    }
```

Comment:
I went with part of the suggestions here, but I'll explain why I left some of them:
- I don't really agree with abstracting textures as "chunks", "fragments" or "parts", since they are treated as textures in the implementation
- For consistency with other texture rectangle names mentioned in another comment, I kept `rectLeft`/`rectTop` instead of `x`/`y`, but I agree they should be changed if the other ones end up being changed.
- `totalSizeRemaining` seems confusing to me because it's a mix between "total" and "remaining" size which are different things, so I think I prefer keeping it as `remainingSize`
- I couldn't change all the `<< 2` to `* 4` while keeping it matching unfortunately, so I didn't change any of them for consistency (which also means I couldn't use `G_IM_SIZ_32b_BYTES`)